### PR TITLE
[MOB-89] Separate headers

### PR DESCRIPTION
**Pull request by ameer2468**
**This PR**: Provides and creates a few different headers to accomodate and fullfill different purposes.

- The styling has been duplicated on purpose to allow for design iterations as needed, instead of wrapping all different headers with the same style.
- Instead of manually setting "title" for different routes, now many screens simply use the `route` object for cleaner code, however in some cases like in SettingsStack, title can be provided manually since the names of the `route` themselves are not desired.

I think this is good for now - as it offers the level of flexibility we need to iterate the design of each header + support better understanding of our headers.

### CODE REVIEW

File: apps/mobile/src/components/header/SearchHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		<View
+			style={twStyle('relative h-auto w-full border-b border-app-cardborder bg-app-header', {
+				paddingTop: headerHeight + (isAndroid ? 15 : 0)
+			})}
+		>
+			<View style={tw`mx-auto h-auto w-full justify-center px-5 pb-3`}>
+				<View style={tw`w-full flex-row items-center justify-between`}>
+					<View style={tw`flex-row items-center gap-3`}>
+					<Pressable
+								hitSlop={24}
```
Comment (utkubakir) [inline apps/mobile/src/components/header/SearchHeader.tsx:41]: Let's not use hitSlop unless it's 100% necessary

Comment (utkubakir) [reply apps/mobile/src/components/header/SearchHeader.tsx:41]: eh I guess we can remove them all later, ignoring this for now

**Review by utkubakir — APPROVED** [review]
lgtm

File: apps/mobile/src/components/header/SearchHeader.tsx (hunk 1 above)
Comment (ameer2468) [reply apps/mobile/src/components/header/SearchHeader.tsx:41]: > eh I guess we can remove them all later, ignoring this for now

I added them originally due to Jamie saying buttons were hard to click, and he suggested to increase the radius of pressability (hitSlop).

**Pull request merged**